app/commands/authenticate_user.rb

```diff
@@ -0,0 +1,28 @@
+class AuthenticateUser
+  prepend SimpleCommand
+
+  def initialize username, password
+    @username = username
+    @password = password
+  end
+
+  def call
+    @user = user
+    token = JsonWebToken.encode(user_id: @user.id) if @user
+    @user = ActiveModelSerializers::SerializableResource.new @user
+    {user: @user, auth_token: token}
+  end
+
+  private
+
+  attr_accessor :username, :password
```

Comment:
Sao khai báo này là private ah em

Reply:
để an toàn thôi chị, khai báo thế này thì các instance của class AuthenticateUser không thể getter setter được, ví dụ a.username, a.password

Reply:
muốn access được nó thì chỉ có thể sử dụng biến local, còn biến instance thì không